tests/utils/stl/test/target_info.py

```diff
@@ -0,0 +1,45 @@
+#===----------------------------------------------------------------------===//
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+#===----------------------------------------------------------------------===//
+
+import locale
+
+
+def _test_locale(loc):

```

Comment:
what is this actually doing?

Like why do we need to muck around with locales for the tests.

Reply:
Lit/libcxx enables/disables tests based on the locales available on the platform

Reply:
OK, why do we need to import that behavior? I may be missing something about how this code is derived from llvm's lit.

Reply:
It's not taken from lit. It's taken from the libcxx implementation of a lit testsuite.
In lit tests there are comments that represent requirements for running/excluding certain tests as well as comments which describe the expected result of a test based on a set of features.
One of the features that the lit libcxx tests consider is the locales available on the system.
I chose to maintain as much of the dynamic feature detection as reasonable and hardcoded the rest.
We could choose to completely ignore these embedded lit directives but I felt that if we were going to use lit, we might as well use it as close to the way it was intended to be used as possible.
If I completely ignore this embedded directives feature (and the fact that I didn't have to write the building of tests in CMake) I should probably have just used CTest.